core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -604,7 +664,28 @@ private static class DeleteFileGroup {
       this.files = files;
     }
 
+    public DeleteFile[] filter(long seq) {
```

Comment:
There is no need to use the Stream API (which has overhead) if column stats filtering is disabled.

<img width="2560" alt="image" src="https://github.com/apache/iceberg/assets/6235869/b7b46915-3c6c-4f89-bacd-c156a9135e37">

Reply:
Would the stream could be costly with `useColumnStatsFiltering`? Should we prefer to use for-loop? It seems we have heavy stream usage.

Reply:
I think the `Stream` API makes it easier to read in case we need to iterate through the elements and check if each of them matches. We would need to try this separately and see how much more complicated it would be and what kind of benefits we get.